testing/file_test/test_file.cpp

```diff
@@ -93,9 +96,53 @@ struct SplitState {
   int file_index = 0;
 };
 
+static auto ExtractFilePathFromUri(llvm::StringRef uri)
+    -> std::optional<llvm::StringRef> {
+  static constexpr llvm::StringRef FilePrefix = "file:/";
+  if (!uri.starts_with(FilePrefix)) {
+    return std::nullopt;
+  }
+  return uri.drop_front(FilePrefix.size());
+}
+
+static auto AutoFillDidOpenParams(llvm::json::Object* params,
```

Comment:
A comment seems like it'd be especially helpful here:

```suggestion
// When `FROM_FILE_SPLIT` is used in `text`, populate the value from the split
// matching the `uri`. Only used for `textDocument/didOpen`.
static auto AutoFillDidOpenParams(llvm::json::Object* params,
```

Reply:
Done.